src/main/java/io/openliberty/tools/common/plugins/util/BinaryScannerUtil.java

```diff
@@ -168,7 +172,7 @@ public Set<String> runBinaryScanner(Set<String> currentFeatureSet, List<String>
                 if (scannerException.getClass().getName().equals(PROVIDED_FEATURE_EXCEPTION)) {
                     // The list of features from the app is passed in but it contains conflicts
                     Set<String> conflicts = getFeatures(scannerException);
-                    Set<String> sampleFeatureList = reRunBinaryScanner(allClassesDirectories, logLocation, targetJavaEE, targetMicroProfile);
+                    Set<String> sampleFeatureList = reRunIfFailed ? reRunBinaryScanner(allClassesDirectories, logLocation, targetJavaEE, targetMicroProfile): null;
```

Comment:
I think we should use the `featureList` returned from the first call to the binary scanner instead of `null`.
Same for the other two reruns below.

Reply:
I'm not so sure. If the message is "A working set of features could not be generated due to conflicts in the application’s API usage: A, B, C" you can't say "...suggested features for guidance: A, B, C."

Reply:
For the `FEATURE_CONFLICT_EXCEPTION`, I agree. I will add a comment there.

For `PROVIDED_FEATURE_EXCEPTION`, we actually do not need to add this extra rerun check logic because this exception only occurs if a feature list was passed in, which means we cannot skip the second binary scanner call.

Reply:
I've updated the logic here to remove the rerun check logic.